docs/examples/openiotsdk_examples_software_update.md

```diff
@@ -0,0 +1,141 @@
+# Matter Open IoT SDK Example Device Firmware Upgrade
+
+Matter Open IoT SDK examples can support over-the-air (`OTA`) Device Firmware
+Upgrade (`DFU`) based on Matter-compliant OTA update protocol that uses the
+Matter operational network for querying and downloading a new firmware image.
+
+## Device Firmware Upgrade over Matter OTA cluster
+
+The `DFU` over Matter OTA clusters requires two kinds of nodes:
+
+-   `OTA Provider` is an application that implement OTA provider cluster
+    service. It can respond to the OTA Requestors' queries about available
+    software updates and share the update image with them.
+-   `OTA Requestor` is an application that implement OTA requestor cluster
+    service. It can communicate with the OTA Provider to fetch applicable
+    software updates and apply it.
+
+The last required element is a Matter controller. This application controls both
+nodes and manages the entire software update process.
+
+In the procedure described below, the `OTA Provider` will be a
+[Linux application](../../examples/ota-provider-app/linux/README.md) and the
+Open IoT SDK example with
+[DFU support](openiotsdk_examples#device-firmware-update) will work as the OTA
+Requestor. The [chip-tool](../../examples/chip-tool/README.md) application used
+as Matter controller.
+
+List of `OIS` examples that currently support the `DFU` over Matter:
+
+-   ota-requestor-app
+
+**NOTE**
+
+Remember to setup the
+[OIS network environment](openiotsdk_examples.md#networking-setup) before
+starting the `DFU` over Matter procedure.
+
+### Device Firmware Upgrade over Matter procedure
+
+1.  Navigate to the CHIP root directory.
+
+2.  Build the `OTA Provider` application:
+
+    ```
+    scripts/examples/gn_build_example.sh examples/ota-provider-app/linux out/ota-provider chip_config_network_layer_ble=false
+    ```
+
+    More details about `OTA provider` application can be found
+    [here](../../examples/ota-provider-app/linux/README.md).
+
+3.  Build `chip-tool`:
+
+    ```
+    scripts/examples/gn_build_example.sh examples/chip-tool out/chip-tool
+    ```
+
+    More details about `chip-tool` application can be found
+    [here](../../examples/chip-tool/README.md).
+
+4.  Build `OIS` example application
+
+    ```
+    scripts/examples/openiotsdk_example.sh -v 1 -V 0.0.1 <example_name>
+    ```
+
+    This is first version of the application that can be updated.
+
+5.  Build `OIS` update image
+
+    ```
+    scripts/examples/openiotsdk_example.sh -p out/update -v 2 -V 0.0.2 <example_name>
+    ```
+
+    Pass the build path (`-p out/update`) in this step to not override the
+    example application. In that directory you should find the update image file
+    with `.ota` extension (`chip-openiotsdk-<example_name>-example.ota`).
+
+6.  In `terminal 1`: run `OTA Provider` application in `OIS` network environment
```

Comment:
Should probably say near the start that 3 terminal windows will be required for the test.